packages/devextreme/js/__internal/core/state_manager/utils.ts

```diff
@@ -0,0 +1,108 @@
+export function splitStatePath(statePath: string, separator = '.'): string[] {
+  if (!statePath || typeof statePath !== 'string') {
+    return [];
+  }
+
+  return statePath.split(separator).filter(Boolean);
+}
+
+export function isValidStatePath(statePath: string): boolean {
+  const parts = splitStatePath(statePath);
+  return parts.length >= 2;
+}
+
+export function areEqual(inputA: unknown, inputB: unknown): boolean {
+  function iter(a: unknown, b: unknown, visited = new WeakMap()): boolean {
+    if (a === b) return true;
+
+    if (a === null || b === null || a === undefined || b === undefined) {
+      return a === b;
+    }
+
+    if (typeof a !== 'object' || typeof b !== 'object') {
+      return a === b;
+    }
```

Comment:
At this moment we already know that `a !== b`:
```suggestion
    if (a === null || b === null || a === undefined || b === undefined) {
      return false;
    }

    if (typeof a !== 'object' || typeof b !== 'object') {
      return false;
    }
```

Reply:
Atually, I'd refactor the whole function to make separated object's comparison, like this:

```ts
if (typeof a === 'object' && typeof b === 'object') {
return equalObjects(a,b);
}
```

Reply:
See the [comment](https://github.com/DevExpress/DevExtreme/pull/29697#discussion_r2158613519)